pkg/cmd/api/api.go

```diff
@@ -424,15 +424,30 @@ func parseErrorResponse(r io.Reader, statusCode int) (io.Reader, string, error)
 		return r, "", err
 	}
 
-	var parsedBody struct {
+	type errorMessage struct {
 		Message string
-		Errors  []struct {
-			Message string
-		}
+	}
+	type errorResponse struct {
+		Message string
+		Errors  []errorMessage
```

Comment:
When unmarshaling a field whose value is unknown at compile time (here, we don't know if the type is `[]errorMessage` or `[]string`), it's best to [use `json.RawMessage` as the type](https://golang.org/pkg/encoding/json/#example_RawMessage_unmarshal), and then check whether its 1st character is a `[` or a `"` and with that decide whether you want to decode it to an `errorMessage` type or a `string`.

Reply:
Tottally agree with you @mislav , your approach is way more flexible. I will work on a solution.